### ISSUE TITLE
A parent directory of the root can be accessed

### ISSUE THREAD

**Issue by hayatoito** [iron/staticfile#89]
It looks that `RequestedPath::new()` uses the result of `decode_percents(...)` without any filtering.
That allows a potential access to a parent directory of the Static's `root`. 

For example, the following request might return the contents of `/etc/passwd` file.
```
http://host:port/..%2f..%2f..%2f..%2f..%2f..%2f..%2f..%2f..%2f..%2f..%2f..%2fetc/passwd
```

I guess this behavior is unintentional because this could be an security vulnerability.

**Comment by Hoverbear** [comment]
Hm, I would feel that this is a security risk yes. 

Would you like to make a patch for this?

**Comment by hayatoito** [comment]
Okay. Let me try.

**Comment by Hoverbear** [comment]
You're the best. :)

**Comment by untitaker** [comment]
Released 0.3.1. Thanks @hayatoito!

**Comment by hayatoito** [comment]
My pleasure. Thank you for merging.
